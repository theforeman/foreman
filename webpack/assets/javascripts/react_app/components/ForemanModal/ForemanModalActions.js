import {
  ADD_MODAL,
  SET_MODAL_OPEN,
  SET_MODAL_CLOSED,
  SET_MODAL_START_SUBMITTING,
  SET_MODAL_STOP_SUBMITTING,
} from './ForemanModalConstants';
import { selectModalExists } from './ForemanModalSelectors';

export const addModal = ({ id, isOpen = false, isSubmitting = false }) => (
  dispatch,
  getState
) => {
  if (selectModalExists(getState(), id)) {
    throw new Error(`ForemanModal with ID ${id} already exists`);
  }
  return dispatch({
    type: ADD_MODAL,
    payload: { id, isOpen, isSubmitting },
  });
};

const modalAction = actionType => ({ id }) => (dispatch, getState) => {
  if (!selectModalExists(getState(), id)) {
    throw new Error(
      `${actionType} error: Modal with id '${id}' does not exist`
    );
  }
  return dispatch({
    type: actionType,
    payload: { id },
  });
};

export const setModalStartSubmitting = modalAction(SET_MODAL_START_SUBMITTING);
export const setModalStopSubmitting = modalAction(SET_MODAL_STOP_SUBMITTING);
export const setModalOpen = modalAction(SET_MODAL_OPEN);
export const setModalClosed = modalAction(SET_MODAL_CLOSED);

// Pass in the ForemanModal id here and get bound action creators with the id already plugged in.
export const bindForemanModalActionsToId = ({ id }) => ({
  addModal: () => addModal({ id }),
  setModalOpen: () => setModalOpen({ id }),
  setModalClosed: () => setModalClosed({ id }),
  setModalStartSubmitting: () => setModalStartSubmitting({ id }),
  setModalStopSubmitting: () => setModalStopSubmitting({ id }),
});
