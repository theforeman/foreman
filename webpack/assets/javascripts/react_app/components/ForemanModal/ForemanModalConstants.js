export const SET_MODAL_OPEN = 'SET_MODAL_OPEN';
export const SET_MODAL_CLOSED = 'SET_MODAL_CLOSED';
export const ADD_MODAL = 'ADD_MODAL';
export const SET_MODAL_START_SUBMITTING = 'SET_MODAL_START_SUBMITTING';
export const SET_MODAL_STOP_SUBMITTING = 'SET_MODAL_STOP_SUBMITTING';
