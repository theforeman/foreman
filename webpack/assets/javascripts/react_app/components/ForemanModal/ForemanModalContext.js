import { createContext } from 'react';

// creating context in a separate file to avoid circular imports
export default createContext(null);
