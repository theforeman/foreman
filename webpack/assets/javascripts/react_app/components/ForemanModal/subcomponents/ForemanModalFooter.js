import React from 'react';
import PropTypes from 'prop-types';
import { Modal, Button } from 'patternfly-react';
import { useModalContext } from '../ForemanModalHooks';
import { translate as __ } from '../../../common/I18n';

import SubmitOrCancel from './SubmitOrCancel';

const ForemanModalFooter = props => {
  const childCount = React.Children.count(props.children);
  const { onClose, isSubmitting, id, submitProps } = useModalContext();

  // Render the provided children, or default markup if none given
  const closeButton = childCount === 0 && (
    <Button bsStyle="default" onClick={onClose}>
      {__('Close')}
    </Button>
  );

  const submitOrCancel = childCount === 0 && submitProps && (
    <SubmitOrCancel
      isSubmitting={isSubmitting}
      onCancel={onClose}
      submitProps={submitProps}
      id={id}
    />
  );

  return (
    <Modal.Footer {...props}>
      {props.children}
      {submitOrCancel || closeButton}
    </Modal.Footer>
  );
};

ForemanModalFooter.propTypes = {
  children: PropTypes.node,
};

ForemanModalFooter.defaultProps = {
  children: null,
};

export default ForemanModalFooter;
