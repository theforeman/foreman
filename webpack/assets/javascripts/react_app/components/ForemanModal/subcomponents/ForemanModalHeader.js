import React from 'react';
import PropTypes from 'prop-types';
import { Modal } from 'patternfly-react';
import { useModalContext } from '../ForemanModalHooks';

const ForemanModalHeader = props => {
  const { title } = useModalContext();
  // title will be falsey if its value is the default ''
  // Render the provided children, or default markup if none given
  return (
    <Modal.Header closeButton {...props}>
      {title && <Modal.Title>{title}</Modal.Title>}
      {props.children}
    </Modal.Header>
  );
};

ForemanModalHeader.propTypes = {
  children: PropTypes.node,
};

ForemanModalHeader.defaultProps = {
  children: null,
};

export default ForemanModalHeader;
