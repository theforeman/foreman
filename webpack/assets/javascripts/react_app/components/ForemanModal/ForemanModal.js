import React from 'react';
import { Modal } from 'patternfly-react';
import PropTypes from 'prop-types';
import ModalContext from './ForemanModalContext';
import ForemanModalHeader from './subcomponents/ForemanModalHeader';
import ForemanModalFooter from './subcomponents/ForemanModalFooter';
import { extractModalNodes } from './helpers';

const ForemanModal = props => {
  const {
    id,
    title,
    onClose,
    isOpen,
    children,
    isSubmitting,
    submitProps,
    ...propsToPassDown
  } = props;
  // Extract header and footer from children, if provided
  const { headerChild, footerChild, otherChildren } = extractModalNodes(
    children
  );
  const context = {
    isOpen,
    onClose,
    isSubmitting,
    id,
    title,
    submitProps,
  };

  const defaultHeader = headerTitle =>
    headerTitle ? <ForemanModalHeader /> : null;
  const headerToRender = headerChild || defaultHeader(title);

  const defaultFooter = subProps =>
    Object.keys(subProps).length !== 0 ? <ForemanModalFooter /> : null;
  const footerToRender = footerChild || defaultFooter(submitProps);

  return (
    <ModalContext.Provider value={context}>
      <Modal
        onHide={onClose}
        show={isOpen}
        className="foreman-modal"
        {...propsToPassDown}
      >
        {headerToRender}
        <Modal.Body>{otherChildren}</Modal.Body>
        {footerToRender}
      </Modal>
    </ModalContext.Provider>
  );
};

ForemanModal.propTypes = {
  children: PropTypes.node,
  title: PropTypes.string,
  id: PropTypes.string.isRequired,
  isOpen: PropTypes.bool,
  onClose: PropTypes.func.isRequired,
  isSubmitting: PropTypes.bool,
  submitProps: PropTypes.object,
};

ForemanModal.defaultProps = {
  children: null,
  isOpen: false,
  title: '',
  isSubmitting: false,
  submitProps: {},
};

export default ForemanModal;
