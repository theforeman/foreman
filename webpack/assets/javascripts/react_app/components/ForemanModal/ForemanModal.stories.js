import React from 'react';
import { Button } from 'patternfly-react';
import storeDecorator from '../../../../../stories/storeDecorator';
import ForemanModal from '.';
import { useForemanModal } from './ForemanModalHooks';
import Story from '../../../../../stories/components/Story';

export default {
  title: 'Components|ForemanModal/Props & Children',
  decorators: [storeDecorator],
};

export const withDefaultHeaderFooter = () =>
  // using createElement here so that hooks work in stories
  React.createElement(() => {
    const { setModalOpen } = useForemanModal({ id: 'default' });
    return (
      <Story>
        <Button bsStyle="primary" onClick={setModalOpen}>
          Show Modal
        </Button>
        <ForemanModal id="default" title="I'm a modal!">
          <ForemanModal.Header />
          If you supply a title prop, it will be used as the modal title.
          <ForemanModal.Footer />
        </ForemanModal>
      </Story>
    );
  });

withDefaultHeaderFooter.story = {
  name: 'With default header & footer',
};

export const withNoChildren = () =>
  // using createElement here so that hooks work in stories
  React.createElement(() => {
    const { setModalOpen } = useForemanModal({ id: 'noChildren' });
    return (
      <Story>
        <Button bsStyle="primary" onClick={setModalOpen}>
          Show Modal
        </Button>
        <ForemanModal id="noChildren" title="I'm a modal!" />
      </Story>
    );
  });

withNoChildren.story = {
  name: 'With no children',
};

export const withCustomHeaderFooter = () =>
  React.createElement(() => {
    const { setModalOpen } = useForemanModal({ id: 'custom' });
    return (
      <Story>
        <Button bsStyle="primary" onClick={setModalOpen}>
          Show Modal
        </Button>
        <ForemanModal id="custom">
          <ForemanModal.Header>
            <h3>This is a custom header! :)</h3>
          </ForemanModal.Header>
          You can provide your own {`<ForemanModal.Header>`}
          <ForemanModal.Footer>
            Click the X in the upper right to close
          </ForemanModal.Footer>
        </ForemanModal>
      </Story>
    );
  });

withCustomHeaderFooter.story = {
  name: 'With custom header & footer',
};

export const withUnorderedHeaderFooter = () =>
  React.createElement(() => {
    const { setModalOpen } = useForemanModal({ id: 'unordered' });
    return (
      <Story>
        <Button bsStyle="primary" onClick={setModalOpen}>
          Show Modal
        </Button>
        <ForemanModal id="unordered" title="I'm a custom modal!">
          <div>
            Header and footer will be correctly ordered when rendering, even if
            they are out of order in the markup
          </div>
          <ForemanModal.Footer>This is the footer</ForemanModal.Footer>
          <ForemanModal.Header>
            <h3>This is the header</h3>
          </ForemanModal.Header>
        </ForemanModal>
      </Story>
    );
  });

withUnorderedHeaderFooter.story = {
  name: 'With unordered header & footer',
};

export const withNoCloseButton = () =>
  React.createElement(() => {
    const { setModalOpen } = useForemanModal({ id: 'noClose' });
    return (
      <Story>
        <Button bsStyle="primary" onClick={setModalOpen}>
          Show Modal
        </Button>
        <ForemanModal id="noClose" title="I'm an X-less modal!">
          <ForemanModal.Header closeButton={false} />
          <br />
          Props passed to ForemanModal.Header will be passed down to
          Modal.Header
          <ForemanModal.Footer />
        </ForemanModal>
      </Story>
    );
  });

withNoCloseButton.story = {
  name: 'With no close button',
};

export const withNoFooter = () =>
  React.createElement(() => {
    const { setModalOpen } = useForemanModal({ id: 'noFooter' });
    return (
      <Story>
        <Button bsStyle="primary" onClick={setModalOpen}>
          Show Modal
        </Button>
        <ForemanModal id="noFooter" title="I'm a modal!">
          <ForemanModal.Header />
          This is the modal body. There is no footer.
        </ForemanModal>
      </Story>
    );
  });

withNoFooter.story = {
  name: 'With no footer',
};

export const withNoHeader = () =>
  React.createElement(() => {
    const { setModalOpen } = useForemanModal({ id: 'noHeader' });
    return (
      <Story>
        <Button bsStyle="primary" onClick={setModalOpen}>
          Show Modal
        </Button>
        <ForemanModal id="noHeader">
          If neither a {`<ForemanModal.Header>`} nor a title prop are supplied,{' '}
          <br />
          the modal will have no header.
          <ForemanModal.Footer />
        </ForemanModal>
      </Story>
    );
  });

withNoHeader.story = {
  name: 'With no header',
};

export const withPropsPassedDownViaSpreadSyntax = () =>
  React.createElement(() => {
    const { setModalOpen } = useForemanModal({ id: 'propsPassed' });
    return (
      <Story>
        <Button bsStyle="primary" onClick={setModalOpen}>
          Show Modal
        </Button>
        <ForemanModal id="propsPassed" title="I'm a modal!" myProp="Hi">
          <ForemanModal.Header />
          The inner {`<Modal>`} component will have any props you pass to
          {`<ForemanModal>`}. (Look in the React dev tools for
          &lsquo;myProp&rsquo;)
          <ForemanModal.Footer />
        </ForemanModal>
      </Story>
    );
  });

withPropsPassedDownViaSpreadSyntax.story = {
  name: 'With props passed down via spread syntax',
};
