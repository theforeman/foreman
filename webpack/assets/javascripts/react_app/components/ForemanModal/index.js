import React, { useEffect } from 'react';
import PropTypes from 'prop-types';
import { useSelector, useDispatch } from 'react-redux';
import {
  selectIsModalOpen,
  selectIsModalSubmitting,
  selectModalExists,
} from './ForemanModalSelectors';
import { setModalClosed, addModal } from './ForemanModalActions';
import ForemanModal from './ForemanModal';
import ForemanModalHeader from './subcomponents/ForemanModalHeader';
import ForemanModalFooter from './subcomponents/ForemanModalFooter';
import reducer from './ForemanModalReducer';

export const reducers = { foremanModals: reducer };

const ConnectedForemanModal = props => {
  const { id, title } = props;
  const isOpen = useSelector(state => selectIsModalOpen(state, id));
  const isSubmitting = useSelector(state => selectIsModalSubmitting(state, id));
  const dispatch = useDispatch();
  const onClose = () => dispatch(setModalClosed({ id }));

  const modalExists = useSelector(state => selectModalExists(state, id));

  useEffect(() => {
    if (modalExists) return; // don't add modal if it already exists
    // https://github.com/facebook/react/issues/14920
    dispatch(addModal({ id, isOpen: false, isSubmitting: false }));
  }, [modalExists, id, dispatch]);

  return (
    <ForemanModal
      {...props}
      id={id}
      title={title}
      isOpen={isOpen}
      isSubmitting={isSubmitting}
      onClose={onClose}
    />
  );
};

ConnectedForemanModal.propTypes = {
  id: PropTypes.string.isRequired,
  title: PropTypes.string,
};

ConnectedForemanModal.defaultProps = {
  title: '',
};

// Header and Footer use the provided children, or default markup if none provided

ConnectedForemanModal.Header = ForemanModalHeader;
ConnectedForemanModal.Footer = ForemanModalFooter;

export default ConnectedForemanModal;
