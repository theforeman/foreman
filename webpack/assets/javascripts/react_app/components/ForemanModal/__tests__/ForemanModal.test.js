import React from 'react';
import ForemanModal from '../ForemanModal';
import ForemanModalHeader from '../subcomponents/ForemanModalHeader';
import ForemanModalFooter from '../subcomponents/ForemanModalFooter';
import { testComponentSnapshotsWithFixtures } from '../../../common/testHelpers';

// This file is for unit tests of the ForemanModal component

const headerChild = <ForemanModalHeader>this is the header</ForemanModalHeader>;
const footerChild = <ForemanModalFooter>this is the footer</ForemanModalFooter>;
const modalBody = <div>This is the body</div>;
const onClose = jest.fn();

const fixtures = {
  'should render': {
    id: 'render',
    title: 'Test modal',
    isOpen: true,
    onClose,
  },
  'should render closed': {
    id: 'closed',
    title: 'Test modal',
    isOpen: false,
    onClose,
  },
  'renders when header and footer are supplied': {
    id: 'withSuppliedChildren',
    title: 'Test modal',
    children: [headerChild, modalBody, footerChild],
    isOpen: true,
    onClose,
  },
  'renders default header child when title prop is present': {
    id: 'defaultHeaderChild',
    title: 'Test modal',
    children: [modalBody, footerChild],
    isOpen: true,
    onClose,
  },
  'renders the supplied header child when title prop is not present': {
    id: 'customHeaderChild',
    children: [headerChild, modalBody, footerChild],
    isOpen: true,
    onClose,
  },
  'renders without header when neither <ForemanModal.Header> nor title prop are present': {
    id: 'noHeader',
    children: [modalBody, footerChild],
    isOpen: true,
    onClose,
  },
};

describe('ForemanModal', () => {
  describe('rendering', () => {
    testComponentSnapshotsWithFixtures(ForemanModal, fixtures);
  });
});
