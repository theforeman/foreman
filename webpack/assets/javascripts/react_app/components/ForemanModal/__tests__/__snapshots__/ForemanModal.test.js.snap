// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForemanModal rendering renders default header child when title prop is present 1`] = `
<ContextProvider
  value={
    Object {
      "id": "defaultHeaderChild",
      "isOpen": true,
      "isSubmitting": false,
      "onClose": [MockFunction],
      "submitProps": Object {},
      "title": "Test modal",
    }
  }
>
  <Modal
    animation={true}
    autoFocus={true}
    backdrop={true}
    bsClass="modal"
    className="foreman-modal"
    dialogComponentClass={[Function]}
    enforceFocus={true}
    keyboard={true}
    manager={
      ModalManager {
        "add": [Function],
        "containers": Array [],
        "data": Array [],
        "handleContainerOverflow": true,
        "hideSiblingNodes": true,
        "isTopModal": [Function],
        "modals": Array [],
        "remove": [Function],
      }
    }
    onHide={[MockFunction]}
    renderBackdrop={[Function]}
    restoreFocus={true}
    show={true}
  >
    <ForemanModalHeader />
    <ModalBody
      bsClass="modal-body"
      componentClass="div"
    >
      <div
        key=".0"
      >
        This is the body
      </div>
    </ModalBody>
    <ForemanModalFooter
      key=".1"
    >
      this is the footer
    </ForemanModalFooter>
  </Modal>
</ContextProvider>
`;

exports[`ForemanModal rendering renders the supplied header child when title prop is not present 1`] = `
<ContextProvider
  value={
    Object {
      "id": "customHeaderChild",
      "isOpen": true,
      "isSubmitting": false,
      "onClose": [MockFunction],
      "submitProps": Object {},
      "title": "",
    }
  }
>
  <Modal
    animation={true}
    autoFocus={true}
    backdrop={true}
    bsClass="modal"
    className="foreman-modal"
    dialogComponentClass={[Function]}
    enforceFocus={true}
    keyboard={true}
    manager={
      ModalManager {
        "add": [Function],
        "containers": Array [],
        "data": Array [],
        "handleContainerOverflow": true,
        "hideSiblingNodes": true,
        "isTopModal": [Function],
        "modals": Array [],
        "remove": [Function],
      }
    }
    onHide={[MockFunction]}
    renderBackdrop={[Function]}
    restoreFocus={true}
    show={true}
  >
    <ForemanModalHeader
      key=".0"
    >
      this is the header
    </ForemanModalHeader>
    <ModalBody
      bsClass="modal-body"
      componentClass="div"
    >
      <div
        key=".1"
      >
        This is the body
      </div>
    </ModalBody>
    <ForemanModalFooter
      key=".2"
    >
      this is the footer
    </ForemanModalFooter>
  </Modal>
</ContextProvider>
`;

exports[`ForemanModal rendering renders when header and footer are supplied 1`] = `
<ContextProvider
  value={
    Object {
      "id": "withSuppliedChildren",
      "isOpen": true,
      "isSubmitting": false,
      "onClose": [MockFunction],
      "submitProps": Object {},
      "title": "Test modal",
    }
  }
>
  <Modal
    animation={true}
    autoFocus={true}
    backdrop={true}
    bsClass="modal"
    className="foreman-modal"
    dialogComponentClass={[Function]}
    enforceFocus={true}
    keyboard={true}
    manager={
      ModalManager {
        "add": [Function],
        "containers": Array [],
        "data": Array [],
        "handleContainerOverflow": true,
        "hideSiblingNodes": true,
        "isTopModal": [Function],
        "modals": Array [],
        "remove": [Function],
      }
    }
    onHide={[MockFunction]}
    renderBackdrop={[Function]}
    restoreFocus={true}
    show={true}
  >
    <ForemanModalHeader
      key=".0"
    >
      this is the header
    </ForemanModalHeader>
    <ModalBody
      bsClass="modal-body"
      componentClass="div"
    >
      <div
        key=".1"
      >
        This is the body
      </div>
    </ModalBody>
    <ForemanModalFooter
      key=".2"
    >
      this is the footer
    </ForemanModalFooter>
  </Modal>
</ContextProvider>
`;

exports[`ForemanModal rendering renders without header when neither <ForemanModal.Header> nor title prop are present 1`] = `
<ContextProvider
  value={
    Object {
      "id": "noHeader",
      "isOpen": true,
      "isSubmitting": false,
      "onClose": [MockFunction],
      "submitProps": Object {},
      "title": "",
    }
  }
>
  <Modal
    animation={true}
    autoFocus={true}
    backdrop={true}
    bsClass="modal"
    className="foreman-modal"
    dialogComponentClass={[Function]}
    enforceFocus={true}
    keyboard={true}
    manager={
      ModalManager {
        "add": [Function],
        "containers": Array [],
        "data": Array [],
        "handleContainerOverflow": true,
        "hideSiblingNodes": true,
        "isTopModal": [Function],
        "modals": Array [],
        "remove": [Function],
      }
    }
    onHide={[MockFunction]}
    renderBackdrop={[Function]}
    restoreFocus={true}
    show={true}
  >
    <ModalBody
      bsClass="modal-body"
      componentClass="div"
    >
      <div
        key=".0"
      >
        This is the body
      </div>
    </ModalBody>
    <ForemanModalFooter
      key=".1"
    >
      this is the footer
    </ForemanModalFooter>
  </Modal>
</ContextProvider>
`;

exports[`ForemanModal rendering should render 1`] = `
<ContextProvider
  value={
    Object {
      "id": "render",
      "isOpen": true,
      "isSubmitting": false,
      "onClose": [MockFunction],
      "submitProps": Object {},
      "title": "Test modal",
    }
  }
>
  <Modal
    animation={true}
    autoFocus={true}
    backdrop={true}
    bsClass="modal"
    className="foreman-modal"
    dialogComponentClass={[Function]}
    enforceFocus={true}
    keyboard={true}
    manager={
      ModalManager {
        "add": [Function],
        "containers": Array [],
        "data": Array [],
        "handleContainerOverflow": true,
        "hideSiblingNodes": true,
        "isTopModal": [Function],
        "modals": Array [],
        "remove": [Function],
      }
    }
    onHide={[MockFunction]}
    renderBackdrop={[Function]}
    restoreFocus={true}
    show={true}
  >
    <ForemanModalHeader />
    <ModalBody
      bsClass="modal-body"
      componentClass="div"
    />
  </Modal>
</ContextProvider>
`;

exports[`ForemanModal rendering should render closed 1`] = `
<ContextProvider
  value={
    Object {
      "id": "closed",
      "isOpen": false,
      "isSubmitting": false,
      "onClose": [MockFunction],
      "submitProps": Object {},
      "title": "Test modal",
    }
  }
>
  <Modal
    animation={true}
    autoFocus={true}
    backdrop={true}
    bsClass="modal"
    className="foreman-modal"
    dialogComponentClass={[Function]}
    enforceFocus={true}
    keyboard={true}
    manager={
      ModalManager {
        "add": [Function],
        "containers": Array [],
        "data": Array [],
        "handleContainerOverflow": true,
        "hideSiblingNodes": true,
        "isTopModal": [Function],
        "modals": Array [],
        "remove": [Function],
      }
    }
    onHide={[MockFunction]}
    renderBackdrop={[Function]}
    restoreFocus={true}
    show={false}
  >
    <ForemanModalHeader />
    <ModalBody
      bsClass="modal-body"
      componentClass="div"
    />
  </Modal>
</ContextProvider>
`;
